### PR TITLE
Added media preview

### PR DESCRIPTION
**Pull request by justfrankie**
new workout form:
image

update workout form:
image

### CODE REVIEW

**Comment by justfrankie** [comment]
mobile responsive screens:

image


image

File: client/src/pages/Dashboard/Dashboard.jsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ const Dashboard = () => {
 
   return (
     <div className={dashboardStyles["workout-item-list"]}>
-      {data.map((item, index) => (
+      {data?.map((item, index) => (
```
Comment (hsu-isaac) [inline client/src/pages/Dashboard/Dashboard.jsx]: bump

File: client/src/pages/Profile/ProfileWidget.jsx [hunk 2]
```diff
@@ -2,12 +2,11 @@ import React from "react";
 import profileStyles from "./ProfileWidget.module.css";
 import { useData } from "../../context/DataContext";
 
-// TODO: make this into arrow function and get rid of curly braces
 const ProfileWidget = () => {
   const { data } = useData();
   const profile = data[0];
 
-  return (
+  return !!profile ? (
```
Comment (hsu-isaac) [inline client/src/pages/Profile/ProfileWidget.jsx]: bump

**Comment by justfrankie** [comment]
added video thumbnail preview!:

image
image

File: client/src/components/MeatballMenu.jsx [hunk 3]
```diff
@@ -2,7 +2,7 @@ import React, { useState, useEffect, useRef } from "react";
 import { ReactComponent as MeatballSvg } from "../assets/meatballs-menu.svg";
 import meatballMenuStyles from "./MeatballMenu.module.css";
 import DeleteConfirmationModal from "./DeleteConfirmationModal";
-import EditWorkoutModal from "./EditWorkoutModal";
+import EditWorkoutModal from "./WorkoutSubmitForm/components/EditWorkoutModal";
```
Comment (justfrankie) [inline client/src/components/MeatballMenu.jsx:5]: moved modal code into sub-components folder

File: client/src/components/WorkoutSubmitForm/WorkoutSubmitForm.jsx [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+  };
+
+  const handlePrivacyClick = () => {
+    console.log("Clicked on Privacy Dropdown");
+  };
+
+  return (
+    <>
+      <div className={workoutSubmitFormStyles["new-workout-container"]}>
+        <form
```
Comment (justfrankie) [inline client/src/components/WorkoutSubmitForm/WorkoutSubmitForm.jsx:146]: If we want to further reduce the size of this component, this is where we'd start abstracting the <input> and <textarea> into its own component. 

Although the file is still kinda large (300+ lines), I think it's at its best balance without needing to make more smaller components

File: client/src/components/WorkoutSubmitForm/WorkoutSubmitForm.module.css [hunk 5]
```diff
@@ -44,14 +44,23 @@
...
-  height: 100px;
+  margin: 10px 0;
+  max-width: inherit;
   background: inherit;
   font: 1rem "Nunito Sans", sans-serif;
   color: #636363;
+  border: none;
+  resize: auto;
+  max-width: 100%;
+  min-width: 100%;
```
Comment (justfrankie) [inline client/src/components/WorkoutSubmitForm/WorkoutSubmitForm.module.css:60]: this is to keep the size of the textarea within the bounds of its parent container

File: client/src/components/WorkoutSubmitForm/components/MediaThumbnailPreviewer.jsx
Comment (justfrankie) [inline client/src/components/WorkoutSubmitForm/components/MediaThumbnailPreviewer.jsx:1]: this file contains two code block that uses very similar logic, and can really be broken down to use better DRY principles. However, I think the current solution is simple enough to understand at a glance, and would be easier to debug if any issues arises

File: client/src/components/WorkoutSubmitForm/components/WorkoutActionButtons.jsx
Comment (justfrankie) [inline client/src/components/WorkoutSubmitForm/components/WorkoutActionButtons.jsx:1]: two different styles of buttons, and locations on where they sit at in the form:

POST new workout:
image

EDIT workout:

image

**Review by hsu-isaac — APPROVED** [review]
LGTM

**Pull request merged**